src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.cs

```diff
@@ -239,7 +239,12 @@ protected virtual void ProduceTagsSynchronously(TaggerContext<TTag> context, Doc
             //
             // Taggers that need to be called accurately should override this method to produce
             // results quickly if possible.
-            ProduceTagsAsync(context, spanToTag, caretPosition).Wait(context.CancellationToken);
+            //
+            // We use JTF.Run here as our tag sources use JTF to quickly switch to the UI thread to report their initial
+            // set of tags.  We need JTF.Run here to ensure we don't deadlock if we're waiting on them while they're
+            // doing that notification.  Specifically, the TagSource will report its first set of tags immediately to the 
+            // UI thread using a JTF.Run and a SwitchToMainThread call.  If we do not use JTF here, we will deadlock.
+            this.ThreadingContext.JoinableTaskFactory.Run(() => ProduceTagsAsync(context, spanToTag, caretPosition));
```

Comment:
💡 The only tagger that calls this is the outlining tagger, and it overrides this method. Rather than delegate to `ProduceTagsAsync`, it would be better to just throw `NotSupportedException`.

Reply:
> The only tagger that calls this is the outlining tagger

This is unfortunately not the case.  If someone requests tags through IAccurageTagger.GetAllTags, then we will do the work to produce tags synchronously (which, by default does things asynchronously with a wait). We support this API for the copy/paste scenario so that copy/paste will get all the correct classifications for a full document.

Note: we could consider getting rid of our support for IAccurateTagger *except* for classification and outlining.